django_redis/pool.py

```diff
@@ -61,7 +63,7 @@ def make_connection_params(self, url):
 
         return kwargs
 
-    def connect(self, url):
+    def connect(self, url: str) -> Redis:
```

Comment:
Could you elaborate more?

Reply:
Something like:

```python
from typing import Generic, TypeVar

PoolType = TypeVar('PoolType', covariant=True, bound=redis.connection.ConnectionPool)
RedisType = TypeVar('PoolType', covariant=True, bound=redis.client.Redis)

class ConnectionFactory(Generic[PoolType, RedisType]):
    def connect(self, url: str) -> RedisType: ...
    def get_or_create_connection_pool(self, params: Mapping[str, Any]) -> PoolType: ...
    def get_connection_pool(self, params: Mapping[str, Any]) -> PoolType: ...
```

which would allow someone to either write a mypy plugin or explicitly declare what types fill in the generic types. However, I'm not sure if people would be likely declaring a typed ConnectionFactory since the types will come from the options in `__init__`.